crates/rattler_installs_packages/src/index/package_database.rs

```diff
@@ -50,26 +51,79 @@ pub struct PackageDb {
     /// A cache of package name to version to artifacts.
     artifacts: FrozenMap<NormalizedPackageName, Box<IndexMap<PypiVersion, Vec<ArtifactInfo>>>>,
 
+    /// Some packages might be associated with a specific registry URL
+    artifact_to_index: HashMap<NormalizedPackageName, usize>,
+
     /// Cache to locally built wheels
     local_wheel_cache: WheelCache,
 
     /// Reference to the cache directory for all caches
     cache_dir: PathBuf,
 }
 
+pub struct PackageSource {
+    pub alias: String,
+    pub url: Url,
+}
+
 impl PackageDb {
     /// Constructs a new [`PackageDb`] that reads information from the specified URLs.
     pub fn new(
+        default_source: Url,
+        extra_sources: &[PackageSource],
         client: ClientWithMiddleware,
-        index_urls: &[Url],
+        package_source_map: BTreeMap<NormalizedPackageName, String>,
```

Comment:
Perhaps it would make sense to create a separate object for this behavior? (Separation of concerns)

Something along these lines perhaps? (Nvm the naming)

```rust
let sources = PackageSources::builder();
sources.add_overwrite("Foo-Bar", "https://some-other.index");
sources.add_index_alias("my-cool-index", "http://my-cool.index");
sources.add_overwrite_by_alias("Bar", "my-cool-index").expect("oeeps alias didnt exist")

let package_db = PackageDb::new(
  ..,
  sources.finish(),
  ..
);

// In the package DB we can just do:
let index: Option<&Url> = sources.get("Foo-Bar");
```